### PR TITLE
docs: fix broken Markdown in `github.md`, plus various cleanups

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: docs/github.md [hunk 1]
```diff
@@ -42,51 +44,56 @@ $ jj new main
...
 ## Working in a Git co-located repository
 
-After doing `jj init --git-repo=.`, git will be in
-a [detached HEAD state][detached], which is unusual, as git mainly works with
-branches. In a co-located repository, `jj` isn't the source of truth. But
-Jujutsu allows an incremental migration, as `jj commit` updates the HEAD of the
-git repository.
+After doing `jj init --git-repo=.`, Git will be in
+a [detached HEAD state][detached], which is unusual, as Git mainly works with
+branches. In a co-located repository, `jj` isn't the source of truth; instead,
```
Comment (ilyagr) [inline docs/github.md:71]: Optional: delete the "isn't the source of truth":

> In a co-located repository, every `jj` command will automatically synchronize Jujutsu's view of the repo with Git's view.

File: docs/github.md [hunk 2]
```diff
@@ -105,7 +112,7 @@ top, you can do that by doing something like this:
 ```shell
 $ # Create a new commit on top of the `your-feature` branch from above.
 $ jj new your-feature
-$ # Address the comments, by updating the code
+$ # Address the comments, by updating the code, then review the changes
```
Comment (ilyagr) [inline docs/github.md:115]: There is an extraneous comma. Also, optionally, I'd split this into two lines.

> $ # Address the comments, by updating the code.
> $ # After that, review the changes

File: docs/github.md [hunk 3]
```diff
@@ -125,7 +132,7 @@ achieved without creating a new commit.
...
```
Comment (ilyagr) [inline docs/github.md:135]: Same as above.

File: docs/github.md [hunk 4]
```diff
@@ -142,23 +149,24 @@ something like this:
 
 ```shell
 $ # Create a new commit on top of the second-to-last commit in `your-feature`,
-$ # as reviews requested a fix there.
+$ # as reviewers requested a fix there.
 $ jj new your-feature- # NOTE: the trailing hyphen is not a typo!
-$ # Address the comments by updating the code
-$ # Review the changes
+$ # Address the comments, by updating the code, then review the changes
```
Comment (ilyagr) [inline docs/github.md:154]: Should be consistent with whatever you do above. If you prefer this on a single line, that's fine, but I'm pretty sure the comma before the "by" should't be there. Also, even if these are on a single line, two sentences may be better.

File: docs/github.md [hunk 5]
```diff
@@ -142,23 +149,24 @@ something like this:
...
 $ jj diff
 $ # Squash the changes into the parent commit
 $ jj squash
-$ # Push the updated branch to the remote. Jujutsu automatically makes it a force push
+$ # Push the updated branch to the remote. Jujutsu automatically makes it a
+$ #  force push
```
Comment (ilyagr) [inline docs/github.md:159]: Extraneous space.

File: docs/github.md [hunk 6]
```diff
@@ -142,23 +149,24 @@ something like this:
...
 $ jj git push --branch your-feature
 ```
 
-The hyphen after `your-feature` comes from [revset](https://github.com/martinvonz/jj/blob/main/docs/revsets.md) syntax.
+The hyphen after `your-feature` comes from
```
Comment (ilyagr) [inline docs/github.md:163]: I'd add "the" after "from"

**Review by ilyagr — APPROVED** [review]
Looks great!

I made comments for the first thoughts that popped into my head, but feel free to skip any of them if you don't immediately agree or don't feel like spending the time. They are all very minor.

File: docs/github.md (hunk 2 above)
Comment (martinvonz) [reply docs/github.md]: I dropped the comma and split it into two sentences. I left it one a single line.

**Pull request merged**